compiler/src/iree/compiler/Codegen/Common/CombineLayoutTransformation.cpp

```diff
@@ -198,12 +210,159 @@ foldExtractSliceIntoMapScatter(RewriterBase &rewriter,
+  }
+  auto storeOp = dyn_cast<IREE::Codegen::StoreToMemrefOp>(
+      *mapScatterOp->getUsers().begin());
+  if (!storeOp) {
+    return rewriter.notifyMatchFailure(
+        mapScatterOp,
+        "map_scatter user is not an iree_codegen.store_to_memref op");
+  }
+  Value outputBuffer = storeOp.getTarget();
+
+  // Write the padding values directly into the outputBuffer.
+  rewriter.setInsertionPointAfter(storeOp);
+  Location loc = padOp->getLoc();
+  SmallVector<OpFoldResult> padSrcSizes =
+      tensor::getMixedSizes(rewriter, loc, padOp.getSource());
+  SmallVector<OpFoldResult> ubs =
+      tensor::getMixedSizes(rewriter, loc, padOp.getResult());
+  SmallVector<OpFoldResult> lbs(ubs.size(), rewriter.getIndexAttr(0));
+  SmallVector<DistributionConfig> distConfigs = distributionConfigFn(
+      padOp.getSourceType().getShape(), rewriter.getContext());
+
+  auto innerLoopBuilder = [&](OpBuilder &b, Location loopLoc, ValueRange ivs) {
+    // We need to scatter the padding values according to the existing
+    // mapScatterOp transformation, so clone the transformation into the
+    // loop nest.
+    auto clonedMapScatterOp = cast<MapScatterOp>(b.clone(*mapScatterOp));
+    Block &clonedTransformBody =
+        clonedMapScatterOp.getTransformationRegion().getBlocks().front();
+    // Get a pointer to the YieldOp before inlining the Block.
+    auto yieldOp =
+        cast<IREE::LinalgExt::YieldOp>(clonedTransformBody.getTerminator());
+    rewriter.inlineBlockBefore(&clonedTransformBody, clonedMapScatterOp, ivs);
+    rewriter.eraseOp(clonedMapScatterOp);
+    OpBuilder::InsertionGuard g(b);
+    b.setInsertionPointAfter(yieldOp);
+    // Compute the indices into the outputBuffer, and the if condition to
+    // write the padding values. Padding values must obey the existing mask
+    // of the current mapScatterOp, and also be in the low or high pad
+    // range of the padOp.
+    SmallVector<OpFoldResult> low = padOp.getMixedLowPad();
+    Value writeCond = nullptr;
+    for (auto [l, srcSize, idx] : llvm::zip_equal(low, padSrcSizes, ivs)) {
+      Value lowVal = getValueOrCreateConstantIndexOp(rewriter, loc, l);
+      auto isLowPad = rewriter
+                          .create<arith::CmpIOp>(loc, arith::CmpIPredicate::ult,
+                                                 idx, lowVal)
+                          ->getResult(0);
+      Value highPadStart = getValueOrCreateConstantIndexOp(
+          rewriter, loc, IREE::LinalgExt::addOfrs(b, loopLoc, l, srcSize));
+      auto isHighPad =
+          rewriter
+              .create<arith::CmpIOp>(loc, arith::CmpIPredicate::uge, idx,
+                                     highPadStart)
+              ->getResult(0);
+      Value isPad = rewriter.create<arith::OrIOp>(loc, isLowPad, isHighPad);
+      if (!writeCond) {
+        writeCond = isPad;
+        continue;
+      }
+      writeCond = rewriter.create<arith::OrIOp>(loc, writeCond, isPad);
```

Comment:
optional nit: if it were me, I'd use ternary conditional operator. I'd not use `continue` for this kind of code because I'd start thinking if people would add new code after the assignment or not. It is clearer if we use `if-else`, then I like ternary conditional operator better. It's just my own rule, so I mark it optional.

```suggestion
      writeCond = !writeCond
                      ? isPad
                      : rewriter.create<arith::OrIOp>(loc, writeCond, isPad);
```